bin/handlers/npm-handler.ts

```diff
@@ -1,5 +1,187 @@
 import type { PackageManagerCompletion } from '../package-manager-completion.js';
+import { stripAnsiEscapes, type ParsedOption } from '../utils/text-utils.js';
+import {
+  LazyCommand,
+  OptionHandlers,
+  commonOptionHandlers,
+  setupLazyOptionLoading,
+  setupCommandArguments,
+  safeExec,
+  safeExecSync,
+  createLogLevelHandler,
+} from '../utils/shared.js';
+
+const ALL_COMMANDS_RE = /^All commands:\s*$/i;
+const OPTIONS_SECTION_RE = /^Options:\s*$/i;
+const SECTION_END_RE = /^(aliases|run|more)/i; // marks end of Options: block
+const COMMAND_VALIDATION_RE = /^[a-z][a-z0-9-]*$/;
+const NPM_OPTION_RE =
+  /(?:\[)?(?:-([a-z])\|)?--([a-z][a-z0-9-]+)(?:\s+<[^>]+>)?(?:\])?/gi;
+const ANGLE_VALUE_RE = /<[^>]+>/;
+const INDENTED_LINE_RE = /^\s/;
+
+function toLines(helpText: string): string[] {
+  return stripAnsiEscapes(helpText).split(/\r?\n/);
+}
+
+function readIndentedBlockAfter(lines: string[], headerRe: RegExp): string {
+  const start = lines.findIndex((l) => headerRe.test(l.trim()));
+  if (start === -1) return '';
+
+  let buf = '';
+  for (let i = start + 1; i < lines.length; i++) {
+    const line = lines[i];
+    if (!INDENTED_LINE_RE.test(line) && line.trim() && !line.includes(','))
+      break;
+    if (INDENTED_LINE_RE.test(line)) buf += ' ' + line.trim();
+  }
+  return buf;
+}
+
+const listHandler =
+  (values: string[], describe: (v: string) => string = () => ' ') =>
+  (complete: (value: string, description: string) => void) =>
+    values.forEach((v) => complete(v, describe(v)));
+
+const npmOptionHandlers: OptionHandlers = {
+  ...commonOptionHandlers,
+
+  loglevel: createLogLevelHandler([
+    'silent',
+    'error',
+    'warn',
+    'notice',
+    'http',
+    'info',
+    'verbose',
+    'silly',
+  ]),
+
+  'install-strategy': listHandler(
+    ['hoisted', 'nested', 'shallow', 'linked'],
+    (v) =>
+      (
+        ({
+          hoisted: 'Hoist all dependencies to top level',
```

Comment:
we should never hardcode descriptions when we parse outputs. it should be either loaded from the help or it should not be there.